api4/user.go

```diff
@@ -531,6 +534,58 @@ func revokeSession(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func autocompleteUsers(c *Context, w http.ResponseWriter, r *http.Request) {
+	channelId := r.URL.Query().Get("in_channel")
+	teamId := r.URL.Query().Get("in_team")
+	term := r.URL.Query().Get("name")
+
+	if c.Session.GetTeamByTeamId(teamId) == nil {
+		if !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+			return
+		}
+	}
+
+	if !app.SessionHasPermissionToChannel(c.Session, channelId, model.PERMISSION_READ_CHANNEL) {
+		c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+		return
+	}
+
+	searchOptions := map[string]bool{}
+
+	hideFullName := !utils.Cfg.PrivacySettings.ShowFullName
+	if hideFullName && !app.SessionHasPermissionTo(c.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		searchOptions[store.USER_SEARCH_OPTION_NAMES_ONLY_NO_FULL_NAME] = true
+	} else {
+		searchOptions[store.USER_SEARCH_OPTION_NAMES_ONLY] = true
+	}
+
+	autocomplete, err := app.SearchUsersInChannel(channelId, term, searchOptions)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	for _, p := range autocomplete {
+		sanitizeProfile(c, p)
+	}
+
+	w.Write([]byte(model.UserListToJson(autocomplete)))
+}
+
+func sanitizeProfile(c *Context, user *model.User) *model.User {
```

Comment:
Sanitizing should not be done in the `api` layer. If the endpoint needs sanitization and the `app` functions you're using don't already support it, you need to modify those `app` functions to perform that sanitization. There should be some examples in `app/user.go`